client/components/data/query-jetpack-user-connection/index.jsx

```diff
@@ -1,43 +1,27 @@
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import { requestJetpackUserConnectionData } from 'calypso/state/jetpack/connection/actions';
 import isRequestingJetpackUserConnection from 'calypso/state/selectors/is-requesting-jetpack-user-connection';
 
-class QueryJetpackUserConnection extends Component {
-	static propTypes = {
-		siteId: PropTypes.number,
-		// Connected props
-		isRequesting: PropTypes.bool,
-		requestJetpackUserConnectionData: PropTypes.func.isRequired,
-	};
-
-	componentDidMount() {
-		this.request( this.props );
+const request = ( siteId ) => ( dispatch, getState ) => {
+	if ( ! isRequestingJetpackUserConnection( getState(), siteId ) ) {
+		dispatch( requestJetpackUserConnectionData( siteId ) );
 	}
+};
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.siteId === nextProps.siteId ) {
-			return;
-		}
-		this.request( nextProps );
-	}
+function QueryJetpackUserConnection( { siteId } ) {
+	const dispatch = useDispatch();
 
-	request( props ) {
-		if ( props.siteId && ! props.isRequesting ) {
-			props.requestJetpackUserConnectionData( props.siteId );
-		}
-	}
+	useEffect( () => {
+		dispatch( request( siteId ) );
```

Comment:
Do we need to check whether `siteId` is truthy?

Reply:
I don't think so. In the single usage of this component, we'll run it [only if it's a Jetpack site](https://github.com/Automattic/wp-calypso/blob/a5763e14e03206b78dd183bcbb583280c1110838/client/my-sites/site-settings/manage-connection/site-ownership.jsx#L270) and for that to be true, we must have a site ID.

Reply:
Also, the prop is specified as a number and as a required one, so it should be fine.

Reply:
> In the single usage of this component, we'll run it [only if it's a Jetpack site](https://github.com/Automattic/wp-calypso/blob/a5763e14e03206b78dd183bcbb583280c1110838/client/my-sites/site-settings/manage-connection/site-ownership.jsx#L270) and for that to be true, we must have a site ID.

Makes sense 👍🏻